nsqd/diskqueue.go

```diff
@@ -341,6 +353,10 @@ func (d *diskQueue) writeOne(data []byte) error {
 	d.writePos += totalBytes
 	atomic.AddInt64(&d.depth, 1)
 
+	// TODO: this is off by one, but we cannot fix this without also fixing
+	// readOne above. We cannot fix readOne above until we guarantee that
+	// every old diskqueue format has been fully processed, otherwise we
+	// will miss one message from old formatted files.
```

Comment:
Fair enough, however for all practical purposes I don't think this is something that needs to be addressed other than perhaps the documentation around the config param.

Can we remove this comment?